### ISSUE TITLE
Can't install ruby on OpenBSD

### ISSUE THREAD

**Issue by desnudopenguino** [rbenv/ruby-build#1208]
cc1: error: unrecognized command line option "-Wno-packed-bitfield-compat"
*** Error 1 in ext/openssl (Makefile:305 'ossl_x509store.o': @ gcc -I. -I../../.ext/include/x86_64-openbsd6.3 -I../.././include -I../.././ex...)
*** Error 1 (Makefile:305 'ossl_x509req.o': @ gcc -I. -I../../.ext/include/x86_64-openbsd6.3 -I../.././include -I../.././ext/openssl  -DRUBY...)
*** Error 1 (Makefile:305 'ossl_x509revoked.o': @ gcc -I. -I../../.ext/include/x86_64-openbsd6.3 -I../.././include -I../.././ext/openssl  -D...)
*** Error 1 (Makefile:305 'openssl_missing.o': @ gcc -I. -I../../.ext/include/x86_64-openbsd6.3 -I../.././include -I../.././ext/openssl  -DR...)
*** Error 1 in target 'openssl_missing.o'
*** Error 1 in target 'ossl_x509revoked.o'
*** Error 1 in target 'ossl_x509req.o'
*** Error 2 in . (exts.mk:231 'ext/openssl/all')
*** Error 2 in /tmp/ruby-build.20180610214248.99813/ruby-2.4.1 (Makefile:775 'build-ext')

ruby-build 20180601
OpenBSD 6.3

I tried installing ruby 2.4.1 and 2.5.0 with the same results

[full log](https://gist.github.com/desnudopenguino/c050bec601d95ee7075e140ec973d278)

**Comment by desnudopenguino** [comment]
Updated the full log. found that a more recent version of gcc (4.9.4) is found in OpenBSD's egcc package. Still having issues with OpenSSL compiling though. I tried to use `CC=egcc CXX=eg++ AUTOCONF_VERSION=2.69 RUBY_CONFIGURE_OPTS=--with-openssl-dir=/usr/local/include/eopenssl/openssl rbenv install 2.4.4` to install ruby this last time, and it *appears* that it is trying to use /usr/include/openssl as the openssl directory regardless of my using the --with-openssl-dir option.

**Comment by p-mongo** [comment]
Trying to debug ruby build failures through rbenv is often difficult. Are you able to build ruby from its official tarballs on this machine?

**Comment by xaque208** [comment]
Reverting this commit locally allows the `./configure` to continue.  https://github.com/rbenv/ruby-build/commit/d155b41eb051f05bb172fa8fddcd0a04a6bcff3f

What I get stuck on next is what looks like openssl missing.

```cc1: error: unrecognized command line option "-Wno-packed-bitfield-compat"
*** Error 1 in ext/openssl (Makefile:305 'ossl_x509store.o': @ gcc -I. -I../../.ext/include/x86_64-openbsd6.3 -I../.././include -I../.././ex...)
*** Error 1 (Makefile:305 'openssl_missing.o': @ gcc -I. -I../../.ext/include/x86_64-openbsd6.3 -I../.././include -I../.././ext/openssl  -DR...)
*** Error 1 in target 'openssl_missing.o'
*** Error 2 in . (exts.mk:236 'ext/openssl/all')
*** Error 2 in /tmp/ruby-build.20180820133140.76654/ruby-2.4.4 (Makefile:775 'build-ext')
zsh: exit 1```

**Comment by mislav** [comment]
Thank you for reporting! Fixed in 48be7c6d76fcf094172bce8df89529d40131667b

**Comment by xaque208** [comment]
Thank you!
